libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -767,6 +766,38 @@ void OverrideChecker::checkOverrideList(OverrideProxy _item, OverrideProxyBySign
 			}
 		}
 	}
+	else if (specifiedContracts.size() > 1)
```

Comment:
Wouldn't this be much easier, if you did it *after* you checked that the specified contracts match the expected contracts (and only if they match)?

Then you could just do something like this, couldn't you?
```
auto overrideIterator = _item.overrides()->overrides().begin();
for (auto const* expected: _item.contract().annotation.linearizedBaseContracts | boost::adaptor::filter([](auto const* base) -> bool { return expectedContracts.count(base); })
  if ((overrideIterator++)->annotation().referencedDeclaration != expected)
  {
     // warn
     break;
  }
```